lib/workers/branch/index.ts

```diff
@@ -339,7 +344,119 @@ export async function processBranch(
       getAdminConfig().trustLevel === 'high' &&
       is.nonEmptyArray(allowedPostUpgradeCommands)
     ) {
-      for (const upgrade of config.upgrades) {
+      const executionMode =
```

Comment:
@rarkins I think this should moved /refacted  to a new file before this pr to be more maintainable. This file is already very big.